test/controllers/api/changesets_controller_test.rb

```diff
@@ -267,1035 +292,1086 @@ def test_close_not_found
+             <member type='relation' role='some' ref='#{other_relation.id}'/>
+            </relation>
+           </modify>
+          </osmChange>
 CHANGESET
 
-      # upload it
-      post :upload, :params => { :id => changeset_id }, :body => diff
-      assert_response :success,
-                      "can't upload a simple valid diff to changeset: #{@response.body}"
-
-      # check that the changes made it into the database
-      assert_equal 0, Node.find(node.id).tags.size, "node #{node.id} should now have no tags"
-      assert_equal 0, Way.find(way.id).tags.size, "way #{way.id} should now have no tags"
-      assert_equal 0, Relation.find(relation.id).tags.size, "relation #{relation.id} should now have no tags"
+        # upload it
+        post :upload, :params => { :id => changeset_id, :api_version => api_version }, :body => diff
+        assert_response :success,
+                        "can't upload a simple valid diff to changeset: #{@response.body}"
+
+        # check that the changes made it into the database
+        assert_equal 0, Node.find(node.id).tags.size, "node #{node.id} should now have no tags"
+        assert_equal 0, Way.find(way.id).tags.size, "way #{way.id} should now have no tags"
+        assert_equal 0, Relation.find(relation.id).tags.size, "relation #{relation.id} should now have no tags"
+      end
     end
 
     ##
     # upload something which creates new objects using placeholders
     def test_upload_create_valid
-      user = create(:user)
-      changeset = create(:changeset, :user => user)
-      node = create(:node)
-      way = create(:way_with_nodes, :nodes_count => 2)
-      relation = create(:relation)
-
-      basic_authorization user.email, "test"
-
-      # simple diff to create a node way and relation using placeholders
-      diff = <<CHANGESET.strip_heredoc
-        <osmChange>
-         <create>
-          <node id='-1' lon='0' lat='0' changeset='#{changeset.id}'>
-           <tag k='foo' v='bar'/>
-           <tag k='baz' v='bat'/>
-          </node>
-          <way id='-1' changeset='#{changeset.id}'>
-           <nd ref='#{node.id}'/>
-          </way>
-         </create>
-         <create>
-          <relation id='-1' changeset='#{changeset.id}'>
-           <member type='way' role='some' ref='#{way.id}'/>
-           <member type='node' role='some' ref='#{node.id}'/>
-           <member type='relation' role='some' ref='#{relation.id}'/>
-          </relation>
-         </create>
-        </osmChange>
+      all_api_versions.each do |api_version|
+        user = create(:user)
+        changeset = create(:changeset, :user => user)
+        node = create(:node)
+        way = create(:way_with_nodes, :nodes_count => 2)
+        relation = create(:relation)
+
+        basic_authorization user.email, "test"
+
+        # simple diff to create a node way and relation using placeholders
+        diff = <<CHANGESET.strip_heredoc
+          <osmChange>
```

Comment:
At one point we'd also need to include a version number in `<osmChange>` (maybe not now)

Reply:
Yeah, I noticed that during the refactoring, but lets leave that for now since we're not planning any changes to the osmchange format (afaik).

Reply:
As I pointed out above, we return the data model version in all kinds of places, not just in OSC format. That needs to be decoupled from the API version.